File: leetcode/2024/cpp/242/isAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> map;
        for (auto a : s)
        {
            map[a] += 1;
        }
        for (auto a : t)
        {
            map[a] -= 1;
        }
        for (auto a: map)
        {
            if (a.second != 0)
            {
                return false;
            }
        }
        return true;
    }
};

File: leetcode/2024/python/242/isAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        m = {}
        for i in s:
            m[i] = m.get(i, 0) + 1
        for i in t:
            if i not in m:
                return False
            m[i] -= 1
        for i in m.values():
            if i != 0:
                return False
        return True
